scipy/stats/tests/test_distributions.py

```diff
@@ -2092,6 +2092,62 @@ def test_logcdf_logsf(self):
         logsf = stats.invgauss.logsf(110, 1.05)
         assert_allclose(logsf, -56.1467092416426)
 
+    def test_pdf(self):
+        # regression tests for improvements introduces in gh-13665
+        # these tests are bases on Gyner & Smyth (2016)
+
+        # infinite mean corresponds to an inverse-chisquare pdf.
+        assert_allclose(
+            stats.invgauss.pdf([-1, 0, 1, 2, np.inf, np.nan], mu=np.inf),
+            [0., 0., 0.24197072, 0.10984782, 0., np.nan],
+            atol=1e-08
+        )
+        # extreme values have zero density for any finite mean value
+        assert_allclose(stats.invgauss.pdf([-1, 0, np.inf], mu=1), [0, 0, 0])
+
+    def test_ppf_sf(self):
+        p = [0.000001, 0.00001, 0.0001, 0.001, 0.01, 0.1, 0.5, 0.9, 0.99,
+             0.999, 0.9999, 0.99999, 0.999999]
+        p1 = stats.invgauss.cdf(stats.invgauss.ppf(p, mu=1), mu=1)
+        abs_error = np.abs(p - p1)
+        # five-number summary of the absolute error
+        fns = np.percentile(
+            abs_error, [0, 25, 50, np.mean(abs_error), 75, 100]
+        )
+        assert_allclose(fns, [0., 0., 8.13151629e-20, 0.,
+                        5.55111512e-17, 2.22044605e-16], atol=1e-08)
+
+        q = stats.invgauss.ppf(p, mu=1)
+        q1 = stats.invgauss.ppf(stats.invgauss.cdf(q, mu=1), mu=1)
+        rel_error = np.abs(q1 - q) / q
+        fns2 = np.percentile(
+            rel_error, [0, 25, 50, np.mean(rel_error), 75, 100]
+        )
+        assert_allclose(fns2, [0., 0., 1.17819649e-16, 0.,
+                        3.50412747e-16, 1.68867330e-11], atol=1e-08)
+
+        # tests if algorithm does not diverge for small probabilities.
+        assert np.isclose(stats.invgauss.ppf(0.00013, mu=1 / 3) * 3,
+                          0.15039762631802803, atol=1e-17)
+        # test if it returns right tail values accurately
+        assert np.isclose(stats.invgauss.isf(1e-17, mu=1.05) / 0.7,
+                          105.42079072389444, atol=1e-14)
+        assert np.isclose(stats.invgauss.isf(1e-50, mu=1.05) / 0.7,
+                          339.3560284841935, atol=1e-13)
+        # test if correct  out is returned for boundary values
+        with np.errstate(invalid='ignore'):
+            # because probabilities must be in the interval [0, 1] a
+            # RuntimeError warning is raised when passing NaN to the ppf
+            # function. So its necessary to silence these warnings to avoid
+            # CI build failure.
+            assert_allclose(
+                stats.invgauss.ppf([0, 0.5, 1, 2, np.nan], mu=1),
```

Comment:
I think that the domain of the `ppf` function is enforced by `rv_continuous`. Please check that, and if so, there is no need to check that behavior here. You only need to check that your code is working properly.